src/server/pps/server/kube_event_tail.go

```diff
@@ -0,0 +1,170 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	etcd "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func kubeEventTail(ctx context.Context, coreV1 corev1.CoreV1Interface, namespace string, etcdPrefix string, etcdClient *etcd.Client) {
+	// backoff.RetryUntilCancel(ctx, func() (retErr error) { //nolint:errcheck
```

Comment:
Is there any reason to retain the commented-out code?

Reply:
I think the intent is to have only one event watcher running per cluster, so I imagine this code will be added back in at some point before this is merged.